### ISSUE TITLE
Q: Is it possible to plug-and-play with this?

### ISSUE THREAD

**Issue by srenatus** [second-state/wasmedge_wasi_socket#22]
Please bear with me, just wrapping my mind around how the WasmEdge stack works 😅 

_Right or wrong?_ To build, say, a [kafka client library](https://github.com/kafka-rust/kafka-rust) and use it to create a wasm module that wasmedge can run, I'd have to fork the existing client library, and swap out its internals, have it call the low-level  tcp stream functions provided by this crate.

Thanks in advance!

**Comment by hydai** [comment]
Correct. Actually, we wrap the low-level socket operators as several tcp stream functions. To use the WASI-Socket feature, you will need to use this crate to replace the existing client library. Thanks.
